### PR TITLE
Add DRB reservoir data

### PR DESCRIPTION
**Pull request by limnoliver**
This PR adds the data necessary to model Pepacton (`nhdhr_151957878`) and Cannonsville (`nhdhr_120022743`) reservoirs in the DRB using GLM. I created new "NWIS" processing steps (yamls) in the pipeline anticipating that we would do broader reservoir data pulls in the future. These are are pretty hard coded to these two reservoirs now, but shouldn't be too hard to do a broader call to NWIS. 

[991e07d] Bathymetry data are from Sciencebase, and the resulting HA tables report actual elevations. There are elevations in the HA table above the spillway.

[1a4b653] Temperature observations are from NWIS. From what I can tell, temperature sensors are at 3 fixed elevations in each reservoir, and so depth was calculated as the surface elevation - sensor elevation. This causes a few issues in the processing pipeline. Because surface elevation changes throughout the day, the measurements are from multiple depths. I created a "depth category" variable that represents the mean depth of each sensor for each day, so that reducing single values per reservoir-day-depth was easier. The actual depths are used when reducing to daily values (which takes the measurement closest to noon). 

[5d23856] Daily surface elevation (water level) data are also from NWIS. Note, there are two sources of water level data for each reservoir. One comes from the same site as temperature observations, and exists since 2019. The other is a different site and extends back to 2013. Note, you have to use different methods to retrieve these data from NWIS. The temperature + water level data site only exist in NWISuv, while the historical water level data only exist in NWISdv. For accessing water level data in NWISdv in these reservoirs, you have to use stat code `00011` -- using the default mean stat code will return an empty dataframe. The longer term data sites seem to have an approval lag, so I gap filled recent dates with the surface elevation data from the temperature + elevation sites. 

I did not add secchi data, but looks like these must exist in LAGOS or WQP there are clarity estimates in the nml files for these nhdids. 

[d047628] I kept the final [reservoir temperature data separate](https://github.com/USGS-R/lake-temperature-model-prep/compare/master...limnoliver:drb_reservoir_data?expand=1#diff-bcc19355ec4f9c806276ab0cb5e8a7e8c5a6de7c46166fdd7808459e5591faa6) from all other temperature data. This is because of the added "depth category" column described above and the need to handle that differently when reducing to daily data. 

Other than the standard nml files and temperature data files, the [water level file](https://github.com/USGS-R/lake-temperature-model-prep/compare/master...limnoliver:drb_reservoir_data?expand=1#diff-797a0a673a888cf24255a5aa39213af1b4c6168ba3eb21ea10b47aff5f8ab70e) might be the other piece that's needed to run GLM.

### CODE REVIEW

**Review by aappling-usgs — APPROVED** [review]
Merging so I can start exploring, but also leaving a couple of ideas/questions here.

File: 7a_nwis_munge/src/munge_reservoir_data.R [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    dplyr::select(-c(X_.backup.from.radar._62615_Inst,
+                     X_.backup.from.radar._62615_Inst_cd)) %>%
+    rename(surface_elevation = X_62615_Inst, surface_elevation_cd = X_62615_Inst_cd)
+
+  # annoying problem where surface elevation and temperature taken on same date/time,
+  # but are in different rows, so NAs are populated. Resolved here.
+  dat_levels <- dat %>% dplyr::select(site_no, dateTime, surface_elevation, surface_elevation_cd) %>%
+    filter(!is.na(surface_elevation))
+
+  dat_fixed <-  dplyr::select(dat, -surface_elevation, -surface_elevation_cd) %>%
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 7a_nwis_munge/src/munge_reservoir_data.R:15]: Your understanding is correct, and this chunk does not fix the whole problem. This chunk just matches the rows with surface elevation to the rows with a temperature measurement (the left join), but does not get rid of duplicated surface elevations. This is handled below in the pivot longer, where are all rows with NAs in the temperature column are dropped (see line 25).

File: 7a_nwis_munge/src/munge_reservoir_data.R [hunk 2]
```diff
@@ -0,0 +1,94 @@
+
+munge_nwis_temperature <- function(in_ind, out_ind, xwalk){
+
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    renameNWISColumns() %>%
+    dplyr::select(-c(X_.backup.from.radar._62615_Inst,
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 7a_nwis_munge/src/munge_reservoir_data.R:6]: Fairly certain this was both sites, but will check and comment.

File: 7a_nwis_munge/src/munge_reservoir_data.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (aappling-usgs) [reply 7a_nwis_munge/src/munge_reservoir_data.R:15]: Ahh, OK, thanks

**Pull request merged**